app/controllers/concerns/lock_ingestible_concern.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Logic to lock Ingestible object, to be included in related controllers
+module LockIngestibleConcern
+  extend ActiveSupport::Concern
+
+  def try_to_lock_ingestible
+    return if @ingestible.obtain_lock(current_user)
+
+    redirect_to ingestibles_path,
+                alert: t('.ingestible_locked', user: @ingestible.locked_by_user.name)
```

Comment:
Initially I wanted to redirect to show page, but currently show page renders edit form, so I've redirected to ingestibles list instead

Reply:
Yes. There's basically no scenario where a back-end editor user would want to just view an ingestible. They are interim entities that you work with until they are ready, and then they feed into the main catalog and are archived forever.  

Though I guess we may occasionally want to allow another editor to look at an ingestible (e.g. an admin or a developer to help another editor understand some error or bug), so perhaps a view does make sense to implement later, but I didn't want to spend any time at all on it before we have the designed screens anyway.